tests/serde.rs

```diff
@@ -541,6 +541,28 @@ fn maps_objects_string_key() {
     test_via_json_with_config(src, Serializer::new().serialize_maps_as_objects(true));
 }
 
+#[wasm_bindgen_test]
+fn json_serialize_and_back() {
+    #[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+    struct Struct {
+        a: HashMap<String, ()>,
+        b: Option<i32>,
+    }
+
+    let serializer = Serializer::json_compatible();
+    let x = Struct {
+        a: hashmap! {
+            "foo".to_string() => (),
+            "bar".to_string() => (),
+        },
+        b: None,
+    };
+    let json = js_sys::JSON::stringify(&x.serialize(&serializer).unwrap());
```

Comment:
I was wondering if this could reuse the existing helper like `test_via_json_with_config(...)`, but hard to tell from a quick look. Did you try it?

Reply:
@HKalbasi Just to double-check, did you verify that `test_via_json_with_config` works as expected (not only passes with `Serializer::json_compatible` but also fails with `Serializer::new`)? Just want to make sure we don't have a false positive here.

Reply:
Yes, I tried `Serializer::new` and it failed. This one is even better, for example serializing `{ x: None }` to json with normal serializer will lead to `{ x: undefined }` and then `"{}"`. If we deserialize it back to rust and compare, we will get `{ x: None }` again. But `serde_json` serialize it to `{ "x" : null }` and by comparing strings, we will catch this as well.

Reply:
Great!